src/app/activity/bounty/Create.ts

```diff
@@ -1,145 +1,187 @@
-            try {
-                convertedDueDateFromMessage = BountyUtils.validateDate(dueAtMessageText);
-            } catch (e) {
+        let convertedDueDateFromMessage: Date;
+        do {
+            // TODO: update default date to a reaction instead of text input
+            // TODO: update hardcoded no/skip to a REGEX
+            const dueDateMessage = 
+                'When is the work for this bounty due by?\n' + 
+                'Please enter `UTC` date in format `yyyy-mm-dd`, i.e 2022-01-01`? (type \'no\' or \'skip\' for a default value of 3 months from today)';
+            await guildMember.send({ content:  dueDateMessage});
+            const dueAtMessageText = await DiscordUtils.awaitUserDM(dmChannel, replyOptions);
+
+            if (! (dueAtMessageText.toLowerCase() === 'no' || dueAtMessageText.toLowerCase() === 'skip') ) {
+                try {
+                    convertedDueDateFromMessage = BountyUtils.validateDate(dueAtMessageText);
+                } catch (e) {
+                    Log.warn('user entered invalid date for bounty');
+                    await guildMember.send({ content: 'Please try `UTC` date in format `yyyy-mm-dd`, i.e 2021-08-15' });
+                }
+            } else if (dueAtMessageText.toLowerCase() === 'no' || dueAtMessageText.toLowerCase() === 'skip') {
+                convertedDueDateFromMessage = null;
+                break;
+            }
+
+            if (convertedDueDateFromMessage.toString() === 'Invalid Date') {
                 Log.warn('user entered invalid date for bounty');
                 await guildMember.send({ content: 'Please try `UTC` date in format `yyyy-mm-dd`, i.e 2021-08-15' });
             }
-        } else if (dueAtMessageText.toLowerCase() === 'no' || dueAtMessageText.toLowerCase() === 'skip') {
-            convertedDueDateFromMessage = null;
-            break;
-        }
+        } while (convertedDueDateFromMessage.toString() === 'Invalid Date');
+        const dueAt = convertedDueDateFromMessage ? convertedDueDateFromMessage : BountyUtils.threeMonthsFromNow();
 
-        if (convertedDueDateFromMessage.toString() === 'Invalid Date') {
-            Log.warn('user entered invalid date for bounty');
-            await guildMember.send({ content: 'Please try `UTC` date in format `yyyy-mm-dd`, i.e 2021-08-15' });
-        }
-    } while (convertedDueDateFromMessage.toString() === 'Invalid Date');
-    const dueAt = convertedDueDateFromMessage ? convertedDueDateFromMessage : BountyUtils.threeMonthsFromNow();
+        newBounty = await createDbHandler(
+            createRequest,
+            description,
+            criteria,
+            dueAt,
+            guildMember,
+            null);
+    } else {
+        const owedTo = await DiscordUtils.getGuildMemberFromUserId(createRequest.owedTo, createRequest.guildId);
+        newBounty = await createDbHandler(
+            createRequest,
+            null,
+            null,
```

Comment:
Something I'm realizing now that I've seen this implemented: given that IOUs are going to be behind a separate API endpoint as well, what do you think about separating out the IOU db read and write handlers?